src/test/groovy/graphql/execution/ValuesResolverTest.groovy

```diff
@@ -603,6 +602,243 @@ class ValuesResolverTest extends Specification {
+                        ]).build()
+                                    | CoercedVariables.emptyVariables()
+
+        '[{ a: "abc" }, $var ] [{ a: "abc" }, { a: "xyz", b: 789 }]'
+                    | ArrayValue.newArrayValue()
+                        .values([
+                            buildObjectLiteral([
+                                    a: StringValue.of("abc")
+                            ]),
+                            VariableReference.of("var")
+                        ]).build()
+                                    | CoercedVariables.of("var": [a: "xyz", b: 789])
+
+    }
+
+    def "getArgumentValues: invalid oneOf list input because element contains null value - #testCase"() {
+        given: "schema defining input object"
+        def inputObjectType = newInputObject()
+                .name("oneOfInputObject")
+                .withAppliedDirective(Directives.OneOfDirective.toAppliedDirective())
+                .field(newInputObjectField()
+                        .name("a")
+                        .type(GraphQLString)
+                        .build())
+                .field(newInputObjectField()
+                        .name("b")
+                        .type(GraphQLInt)
+                        .build())
+                .build()
+
+        when:
+        def argument = new Argument("arg", inputArray)
+        def fieldArgumentList = newArgument().name("arg").type(list(inputObjectType)).build()
+        ValuesResolver.getArgumentValues([fieldArgumentList], [argument], variables, graphQLContext, locale)
+
+        then:
+        def e = thrown(OneOfNullValueException)
+        e.message == "OneOf type field 'oneOfInputObject.a' must be non-null."
+
+        where:
+
+        testCase    | inputArray    | variables
+
+        '[{ a: "abc" }, { a: null }]'
+                    | ArrayValue.newArrayValue()
+                        .values([
+                            buildObjectLiteral([
+                                    a: StringValue.of("abc")
+                            ]),
+                            buildObjectLiteral([
+                                    a: NullValue.of()
+                            ]),
+                        ]).build()
+                                    | CoercedVariables.emptyVariables()
+
+        '[{ a: "abc" }, { a: $var }] [{ a: "abc" }, { a: null }]'
+                    | ArrayValue.newArrayValue()
+                        .values([
+                            buildObjectLiteral([
+                                    a: StringValue.of("abc")
+                            ]),
+                            buildObjectLiteral([
+                                    a: VariableReference.of("var")
+                            ]),
+                        ]).build()
+                                    | CoercedVariables.of("var": null)
+
+    }
+
+    def "getArgumentValues: invalid oneOf non-null list input because element contains duplicate key - #testCase"() {
```

Comment:
For fun let's try two combination cases: a non-null list and a list of non-null inputs